packages/html/src/render.ts

```diff
@@ -321,7 +321,86 @@ const listen = (
   };
 };
 
+/**
+ * Converts a React-style CSS object to a CSS string.
+ * Supports vendor prefixes, pixel value shorthand, and comprehensive CSS properties.
+ * @param styleObject - The style object with React-style camelCase properties
+ * @returns A CSS string suitable for the style attribute
+ */
+const styleObjectToCssString = (styleObject: Record<string, any>): string => {
+  return Object.entries(styleObject)
+    .map(([key, value]) => {
+      // Skip if value is null or undefined
+      if (value == null) return "";
+
+      // Convert camelCase to kebab-case, handling vendor prefixes
+      let cssKey = key;
+
+      // CSS custom properties (--*) are case-sensitive and should not be transformed
+      if (!key.startsWith("--")) {
+        // Handle vendor prefixes (WebkitTransform -> -webkit-transform)
+        if (/^(webkit|moz|ms|o)[A-Z]/.test(key)) {
+          cssKey = "-" + key;
```

Comment:
This results in `WebkitTransform` -> `--webkit-transform` with two `--`, I don't think this is needed if the goal is to convert `PascalCase` to single-dashed-lowercase `-kebab-case`